physics/cires_vert_orodis_v1.F90

```diff
@@ -0,0 +1,1037 @@
+! subroutine ugwp_taub_oro  
+! subroutine ugwp_oro_lsatdis  
+!
+     subroutine ugwp_drag_mtb( iemax,  nz,                              &
+        elvpd, elvp, hprime , sigma, theta, oc,  oa4, clx4, gam, zpbl,  &
+        up, vp, tp, qp, dp, zpm, zpi, pmid, pint, idxzb, drmtb,taumtb)
+
+      use ugwp_common_v1,  only  : bnv2min, grav, grcp, fv, rad_to_deg, dw2min, velmin, rdi
+      use ugwp_oro_init_v1, only : nridge, cdmb, fcrit_mtb, frmax, frmin, strver
+      
+      implicit none   
+!========================
+! several versions for drmtb => high froude mountain blocking
+! version 1 => vay_2018 ; 
+! version 2 => kdn_2005 ; Kim & Doyle in NRL-2005
+! version 3 => ncep/gfs-2017 -gfs_2017 with lm1997
+!======================== 
+!     real, parameter   ::  Fcrit_mtb  = 0.7   
+      
+      integer, intent(in)              ::  nz     
+      integer, intent(in)              ::  iemax    ! standard ktop  z=elvpd  + 4 * hprime
+      real  , intent(out)              ::  taumtb    
+          
+      integer , intent(out)            ::  idxzb           
+      real, dimension(nz), intent(out) ::  drmtb
+
+      real, intent(in)  ::  elvp, elvpd              !elvp  = min (elvpd  + sigfac * hprime(j), hncrit=10000meters)
+      real, intent(in)  ::  hprime , sigma, theta, oc,  oa4(4), clx4(4), gam
+      real, intent(in)  ::  zpbl     
+
+      real, dimension(nz),    intent(in) :: up, vp, tp, qp, dp, zpm, pmid
+      real, dimension(nz+1),  intent(in) :: zpi, pint
+!
+      real, dimension(nz+1)              :: zpi_zero
+      real, dimension(nz)                :: zpm_zero
+      real ::  vtj, rhok, bnv2, rdz, vtkp, vtk, dzp
+      
+      real, dimension(nz) ::  bn2, uds, umf, cosang, sinang
+      
+      integer :: k,  klow, ktop, kpbl
+      real    :: uhm, vhm, bn2hm, rhohm,                  &
+                 mtb_fix, umag, bnmag, frd_src,           &
+                 zblk, who_iz_normal, rlm97,              &
+                 phiang, ang, pe, ek,                     &
+                 cang, sang, ss2, cs2, zlen,  dbtmp,      &
+                 hamp, bgamm, cgamm      
+  
+!==================================================
+!
+!     elvp + hprime <=>elvp + nridge*hprime, ns =2
+!                 ns = sigfac
+!     tau_parel & tau_normal along major "axes" 
+!
+!  options to block the "flow", choices for [klow, ktop]
+!
+!  1-directional (normal)  & 2-directional "blocking"
+!
+!==================================================
+! no - blocking: drmtb(1:nz) = 0.0
+!=================
+      idxzb = -1
+      drmtb(1:nz) = 0.0
+      taumtb      = 0.0
+      klow = 2
+
+      ktop = iemax
+      hamp = nridge*hprime
+      
+! reminder:        cdmb = 4.0 * 192.0/float(imx)*cdmbgwd(1)   Lellipse= a/2=sigma/hprime
+
+      mtb_fix = cdmb*sigma/hamp        !hamp  ~ 2*hprime and 1/sigfac = 0.25 is inside 1/hamp
+
+      if (mtb_fix == 0.) then
+        print *, cdmb, sigma, hamp
+        print *,  ' MTB == 0'
+           stop
```

Comment:
CCPP schemes should not stop the model.

Reply:
I changed these lines to:
      ! if (mtb_fix == 0.) then
      !    write(errmsg,'(*(a))') cdmb, sigma, hamp, ' MTB == 0'
      !    errflg = 1
      !    return
      ! endif

The lines are commented out for now as this subroutine is not called in the current unified_ugwp scheme.  It will either be deleted, or if it used in the final ugwp_v1 code, we will uncomment these lines out and pass the errmsg and errflg variables in to the subroutine.